msbuild/Xamarin.iOS.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs

```diff
@@ -236,6 +246,84 @@ bool Compile (PDictionary plist)
 			return !Log.HasLoggedErrors;
 		}
 
+		void SetRequiredArchitectures (PDictionary plist)
+		{
+			PObject capabilities;
+
+			if (plist.TryGetValue (ManifestKeys.UIRequiredDeviceCapabilities, out capabilities)) {
+				if (capabilities is PArray) {
+					var architectureValues = new HashSet<string> (new[] { "armv6", "armv7", "arm64" });
+					var array = (PArray) capabilities;
+
+					// Remove any architecture values
+					for (int i = 0; i < array.Count; i++) {
+						var value = array[i] as PString;
+
+						if (value == null || !architectureValues.Contains (value.Value))
+							continue;
+
+						array.RemoveAt (i);
+					}
+
+					// If-and-only-if the TargetArchitecture is a single architecture, set it as a required device capability
+					switch (architectures) {
+					case TargetArchitecture.ARM64:
+						array.Add (new PString ("arm64"));
+						break;
+					case TargetArchitecture.ARMv7:
```

Comment:
What happens if the target architecture is ARMv7s or ARMv7k (watchOS)?

Reply:
I generated an archive with Xcode, this is an iOS + watchOS app: https://www.dropbox.com/s/0q8uibpqaygl14m/PR2409.zip?dl=0

Only the iOS app gets a `UIRequiredDeviceCapabilities` (arm64) from Xcode (forced, the Info.plist had armv7).

watchOS doesn't get anything, Xcode doesn't force any value.

Reply:
For what it's worth: the watchOS extension can't contain `UIRequiredDeviceCapabilities`

```
Unsupported Key. The Info.plist of bundle MyApp.app/Watch/MyApp WatchKit App.app/PlugIns/MyApp WatchKit Extension.appex may not contain the UIRequiredDeviceCapabilities key.
```

Reply:
The WatchApp doesn't add anything for `UIRequiredDeviceCapabilities` but doesn't remove anything either.